code/controllers/subsystem/shuttles.dm

```diff
@@ -213,6 +232,85 @@ var/datum/subsystem/shuttle/SSshuttle
 			return 2
 	return 0	//dock successful
 
+/datum/subsystem/shuttle/proc/requestTransitZone(obj/docking_port/mobile/M)
+	// First, determine the size of the needed zone
+	var/transit_width = M.width + 14 // view range is 7, both ways is 14
+	var/transit_height = M.height + 14
+	// Then find a place to put the zone
+
+	world << "Transit requested for [M]"
+
+	CHECK_TICK
+	var/list/availible = transit_turfs - transit_turfs_in_use
+	CHECK_TICK
+
+	world << "There are [availible.len] turfs that are free"
+
+	var/list/proposed_zone
+	for(var/i in shuffle(availible))
+		var/turf/topleft = i
+		world << "Trying [COORD(topleft)]"
+		CHECK_TICK
+		var/turf/bottomright = locate(topleft.x + transit_width,
+			topleft.y + transit_height, topleft.z)
+		if(!bottomright)
+			continue
+		CHECK_TICK
+		if(!(bottomright in transit_turfs))
+			world << "Bottomright [COORD(bottomright)] not in \
+				transit_turfs, trying another topleft"
+			continue
+		CHECK_TICK
```

Comment:
you could make these in checks much much much faster by either:

* Using areas instead of makers, so you could just check area = /area/space/transit (or some shti)
* Making the transit_turfs list be associated `for (var/T in block()) transit_turfs[T] = T` and then just using `if (!transit_turfs[T])`